enhancements/update/cvo-log-level-api.md

```diff
@@ -0,0 +1,205 @@
+* As an OpenShift administrator, I want to decrease the log level of the CVO to save up more storage space.
+* As an OpenShift engineer, I want to decrease the log level of the CVO for shipped releases, so that the customers don't receive debug logs.
+* As an OpenShift engineer, I want to increase the log level of the CVO for the CI runs, so that I can more easily troubleshoot any potential issues that occurred.
+
+### Goals
+
+Add a user-facing API for controlling the run-time verbosity of the [CVO](https://github.com/openshift/cluster-version-operator).
+
+### Non-Goals
+
+Change the default logging verbosity of the Cluster Version Operator in production OCP clusters.
+
+## Proposal
+
+This enhancement proposes to add the field `OperatorLogLevel` to the structure [`ClusterVersionSpec`](https://github.com/openshift/api/blob/c3f7566f6ef636bb7cf9549bf47112844285989e/config/v1/types_cluster_version.go#L40-L96) that is used in the structure [`ClusterVersion`](https://github.com/openshift/api/blob/c3f7566f6ef636bb7cf9549bf47112844285989e/config/v1/types_cluster_version.go#L18-L34) 
+that represents the configuration for the CVO.
+
+This new field is named exactly after the existing corresponding field [`OperatorLogLevel`](https://github.com/openshift/api/blob/36ce464529eb357673342c06be5886c5463cfc50/operator/v1/types.go#L64-L71) that is used inside the structure [`OperatorSpec`](https://github.com/openshift/api/blob/36ce464529eb357673342c06be5886c5463cfc50/operator/v1/types.go#L49-L89) by other operators. 
+
+Its data type will be the same as the data type used by the Cluster Operators (type [`LogLevel`](https://github.com/openshift/api/blob/36ce464529eb357673342c06be5886c5463cfc50/operator/v1/types.go#L91-L107)). This is to ensure consistency across the operators. However, so as to not cause import cycles, this type will need to be redefined in the 
+[api/config/v1/types_cluster_version.go](https://github.com/openshift/api/blob/master/config/v1/types_cluster_version.go) file.
+
+This enhancement is thus proposing to modify the [api/config/v1/types_cluster_version.go](https://github.com/openshift/api/blob/master/config/v1/types_cluster_version.go) file as described by the following text.
+
+A new field `OperatorLogLevel` will be added to the [`ClusterVersionSpec`](https://github.com/openshift/api/blob/c3f7566f6ef636bb7cf9549bf47112844285989e/config/v1/types_cluster_version.go#L40-L96) structure:
+
+```go
+type ClusterVersionSpec struct {
+	...
+	// operatorLogLevel controls the logging level of the cluster version operator.
+	//
+	// Valid values are: "Normal", "Debug", "Trace", "TraceAll".
+	// Defaults to "Normal".
+	// +optional
+	// +kubebuilder:default=Normal
+	OperatorLogLevel CVOLogLevel `json:"operatorLogLevel,omitempty"`
+}
+```
+
+And a new enum will be defined:
+
+```go
+// +kubebuilder:validation:Enum="";Normal;Debug;Trace;TraceAll
+type CVOLogLevel string
```

Comment:
I would prefer we use the same types as https://github.com/openshift/api/blob/36ce464529eb357673342c06be5886c5463cfc50/operator/v1/types.go#L92 I do not see any value changing the names when we are doing the exact same thing.

Reply:
I have made the reasoning more explicit in the new commit, see https://github.com/openshift/enhancements/blob/a99c2f56d31104ef9f4d56ac4f6a0b4830f2c713/enhancements/update/cvo-log-level-api.md?plain=1#L53-L56

I have also added an alternative to the proposed solution, see https://github.com/openshift/enhancements/blob/a99c2f56d31104ef9f4d56ac4f6a0b4830f2c713/enhancements/update/cvo-log-level-api.md?plain=1#L199-L205